rclpy/rclpy/client.py

```diff
@@ -140,8 +138,7 @@ def service_is_ready(self) -> bool:
 
         :return: ``True`` if a server is ready, ``False`` otherwise.
         """
-        with self.handle as capsule:
-            return _rclpy.rclpy_service_server_is_available(capsule)
+        return self.__client.service_server_is_available()
```

Comment:
@sloretz hmm, why no `with self.__client` context manager here? What if it concurs with `Client.destroy()`?

Reply:
Oops, I started out thinking I'd replace the context manager with a check in C++, but never implemented it. I added back the `with` statements in 5ba08d2e15879313170d7304f285088680ac4ccb